src/sentry/eventstream/kafka/backend.py

```diff
@@ -210,14 +216,25 @@ def handle_shutdown_request(signum: int, frame: Any) -> None:
             i = i + 1
             owned_partition_offsets[key] = message.offset() + 1
 
-            with metrics.timer("eventstream.duration", instance="get_task_kwargs_for_message"):
-                task_kwargs = get_task_kwargs_for_message(message.value())
+            use_kafka_headers = options.get("post-process-forwarder:kafka-headers")
 
-            if task_kwargs is not None:
-                with metrics.timer(
-                    "eventstream.duration", instance="dispatch_post_process_group_task"
-                ):
-                    self._dispatch_post_process_group_task(**task_kwargs)
+            if use_kafka_headers is True:
+                try:
+                    with metrics.timer(
+                        "eventstream.duration", instance="get_task_kwargs_for_message_from_headers"
+                    ):
+                        task_kwargs = get_task_kwargs_for_message_from_headers(message.headers())
+
+                    with metrics.timer(
+                        "eventstream.duration", instance="dispatch_post_process_group_task"
+                    ):
+                        self._dispatch_post_process_group_task(**task_kwargs)
+
+                except Exception:
```

Comment:
I'd add a metric in the exception case as well so you can track that.

Reply:
Good point